markdown/en/documentation/configuration.md

```diff
@@ -18,39 +38,38 @@ This chapter describes the minimum configuration necessary to search and retriev
 ```
 
-### Attributes
-
----
-#### **url**
-Specify the FACT-Finder endpoint. Please note the context name `/FACT-Finder-7.2`. It is not sufficient to provide the top-level domain like `http://web-components.fact-finder.de/` which will result in an error.
-
-**Note:** Make sure you are using the same protocol (http://, https://) for the `url` attribute and webapp. 
-
-Using `<ff-communication url="http://web-components.fact-finder.de/FACT-Finder-7.2" ...>` on a page hosted via **HTTPS** will result in an error.
-
----
+#### **search-immediate**
+The `search-immediate` attribute ensures, that a search is automatically triggered as soon as `ff-communication` was successfully attached to the dom. The triggered search request uses the current URL parameters. If no parameters are present, it uses `*` as query (search for all products) in conjunction with the other settings. 
 
-#### **version**
-Specify the FACT-Finder version. E.g. 7.2, 7.3 and so forth.
+**NOTE**
 
----
+You don't want the `search-immediate` attribute to be present on your landing (home) page because this would trigger a search every time a user enters your shop. Usually you want to use it only on category pages and the search result page.
 
-#### **channel** 
-Specify the channel name which is used for the search. You can check which channels are available at the FACT-Finder UI.
 
----
+## JavaScript Approach
+While `ff-communication` provides a declarative way to configure the communication, it is also possible to do so directly via JavaScript as followed:
 
-#### **search-immediate**
-`search-immediate` ensures that a search is automatically triggered when web components are loaded successfully. The triggered search request uses the current URL parameters. If no parameters are present, it uses `*` as query (search for all products) in conjunction with the parameters provided by the `ff-communication` element. 
+```js
+document.addEventListener("ffReady", () => { // "ffReady"-event ensures global factfinder object to exist
+    factfinder.communication.globalSearchParameter.url = "http://web-components.fact-finder.de/FACT-Finder-7.2";
+    factfinder.communication.globalSearchParameter.version = "7.2";
+    factfinder.communication.globalSearchParameter.channel = "bergfreunde-co-uk";
+    
+    /* after this minimal configuration, a search can be triggered manual e.g. through */
+    factfinder.communication.FFCommunicationEventAggregator.addFFEvent({
+         type: "search",
+         query: "*"
+    });
+});
+``` 
 
 **NOTE**
+To prevent our custom elements from missing a factfinder response, before they where loaded, dispatching of results to subscribers is cached and deferred until all custom elements have loaded. When all custom elements have loaded, `ff-communication` calls `factfinder.communication.ResultDispatcher.startDispatching()`. If `ff-communication` is not used, `factfinder.communication.ResultDispatcher.startDispatching()` has to be called manually in JavaScript.
```

Comment:
replace "custom elements" with "Web Components" so it's more consistent with the rest of the documentation
... before they're ...
... until all Web Components are ready to receive the response. As soon as Web Components have finished loading `ff-communication` calls ...

Reply:
In this case, we should rename `custom elements` through `Web Components` in [#99](https://github.com/FACT-Finder-Web-Components/documentation/pull/99) as well. @omikron-fp  What do you think?

Reply:
We could do that.
I'm wondering though if there is an understanding out there that maps `Web Components` to the whole thing as a framework. In which case it might be more obvious to distinguish between everything as a whole called `Web Components` and the actual HTML elements called `custom elements`.
Is this a real thing or is this just me?

Reply:
I think that the technology we use is `Web Components` and our library provides `custom elements` everyone can use. Hence I would tend to change most cases to `custom elements` :P 

I will create an own issue for more detailed consideration of that topic and skip it in this pull request